numpy/random/tests/test_random.py

```diff
@@ -209,6 +243,38 @@ def test_repeatability(self):
         res = hashlib.md5(val).hexdigest()
         assert_(tgt[np.dtype(np.bool).name] == res)
 
+    def test_int64_uint64_corner_case(self):
+        # When stored in Numpy arrays, `lbnd` is casted
+        # as np.int64, and `ubnd` is casted as np.uint64.
+        # Checking whether `lbnd` >= `ubnd` used to be
+        # done solely via direct comparison, which is incorrect
+        # because when Numpy tries to compare both numbers,
+        # it casts both to np.float64 because there is
+        # no integer superset of np.int64 and np.uint64. However,
+        # `ubnd` is too large to be represented in np.float64,
+        # causing it be round down to np.iinfo(np.int64).max,
+        # leading to a ValueError because `lbnd` now equals
+        # the new `ubnd`.
+
+        dt = np.int64
+        tgt = np.iinfo(np.int64).max
+        lbnd = np.int64(np.iinfo(np.int64).max)
+        ubnd = np.uint64(np.iinfo(np.int64).max + 1)
+
+        # None of these function calls should
+        # generate a ValueError now
+        actual = np.random.randint(lbnd, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint(lbnd, [ubnd] * 3, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, [ubnd] * 3, dtype=dt)
```

Comment:
Disagreement is great, but I think disagreeing about how to handle large ints is not something that should be hidden away in the diff comments for a PR titled "broadcasting" ;). I'd be much happier with a PR that was obviously about that, so that it draws input from the people with better-informed opinions than I.

The scalar test you do here already fails on master, so there's no reason that broadcasting needs to support this to be merged - no compatibility is broken.

> Also, why should we complicate the API even further by allowing wrapping?

Because it comes for free as a result of unsigned integer wraparound

> I would not really count on getting much support for.

[I'm not the first to propose this](https://github.com/numpy/numpy/issues/6745#issuecomment-168659982)